pkg/extension/opampextension/opamp_agent.go

```diff
@@ -448,7 +449,8 @@ func (o *opampAgent) saveEffectiveConfig(dir string) error {
 		}
 		o.logger.Info("Loading Configuration to Validate...")
 
-		_, errValidate := loadConfigAndValidateWithSettings(factories, otelcol.ConfigProviderSettings{
+		//lint:ignore SA1019 we need this deprecated function
+		_, errValidate := otelcoltest.LoadConfigAndValidateWithSettings(factories, otelcol.ConfigProviderSettings{
```

Comment:
Why not using `LoadConfigAndValidate`?

Reply:
OpAmp extension wishes to use its own settings, as we don't want the httpprovider or envprovider factories.